tutorial/marine/quick_start.md

```diff
@@ -0,0 +1,296 @@
+
+```toml
+# Cargo.toml
+[package]
+name = "greeter"
+version = "0.1.0"
+authors = ["boneyard93501 <4523011+boneyard93501@users.noreply.github.com>"]
+edition = "2018"
+description = "greeter, a Marine wasi module"
+license = "Apache-2.0"
+
+[[bin]]    <-- 1
+name = "greeter"
+path = "src/main.rs"
+
+[dependencies] <-- 2
+fluence = { version="0.6.9", features=["logger"]}
+log = "0.4.14"
+
+[dev-dependencies]  <-- 3
+fluence-test = "0.1.9"
+
+[dev]
+[profile.release]
+opt-level = "s"
+```
+
+The Config.toml file provides the meta data necessary to properly instrument a Wasm IT module for execution by the Marine runtime.
+
+```toml
+# Config.toml
+modules_dir = "artifacts/"   <-- 1
+
+[[module]]  <-- 2
+    name = "greeter"
+    mem_pages_count = 1
+    logger_enabled = true
+```
+
+As illustrated in Figure 1 above, the Config.toml file is required for the local execution of Wasm modules such as the Repl or unittests. A Config.toml files consists of:
+
+1. Location of the wasm module to be executed by Marine
+2. One or mode module specific runtime requirements such as module name, logging particulars, local access and memory allocation, Recall, [Wasm memory](https://webassembly.github.io/spec/core/exec/runtime.html#memory-instances) is allocated in linear pages of 64Ki each.
+
+As evident by the existence of the main.rs file, Marine Wasm modules are created as binaries and as we see below, the Marine toolchain compiles our Rust code to the desired wasm32-wasi target. Due to our the Wasm IT target, our Rust files look a little different than Rust application or even wasm-bindgen code.
+
+```rust
+// main.rs
+use fluence::{marine, module_manifest, WasmLoggerBuilder};  <-- 1
+
+module_manifest!();  <-- 2
+
+pub fn main() {
+    WasmLoggerBuilder::new().build().unwrap();  <-- 3
+}
+
+#[marine]  <-- 4
+pub fn greeting(name: String) -> String {  <-- 5
+    format!("Hi, {}", name)
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::greeting;
+    use fluence_test::marine_test; <-- 6
+
+    #[marine_test(config_path = "../Config.toml", modules_dir = "../artifacts")]
+    fn test_greeting() {
+        let name = "Marine".to_string();
+        let res = greeting(name.clone());
+        assert_eq!(res, format!("Hi, {}", name));
+    }
+}
+```
+
+1. We import the necessary dependencies from the fluence crate
+2. The module_manifest macro provides the module version, if needed
+3. Logging is available as via this Fluence crate, which is a feature we need to turn on in our Cargo.toml, if desired
+4. Macro enabling and enforcing our wasm32-wasi types for export as module API
+5. Implementation of exposed, callable Wasm module function
+6. Marine-Test macro to allow unit testing with the (standard) cargo test tool
+
+Now that we got our code and metadata files in place, it's finally time to build our greeting module. Let's start with running the unit test:
+
+```rust
+root@56892f4726bb:/workspaces/devcontainer/greeter# cargo test
```

Comment:
I think it's better to tech people to use `--release` for marine_tests:
```
cargo test --release
```

That's because in debug mode marine loads wasms damn slow. Compare:
Debug:
```
test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 5.46s
```

Release
```
test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.09s
```

It's not so important in greeting example since 5 seconds is OK, but if wasm module becomes x10 bigger, it will take forever.